include/fiction/algorithms/physical_design/apply_gate_library.hpp

```diff
@@ -146,9 +233,57 @@ CellLyt apply_gate_library(const GateLyt& lyt)
 
     detail::apply_gate_library_impl<CellLyt, GateLibrary, GateLyt> p{lyt};
 
-    auto result = p.run();
+    try
+    {
+        return p.run();
+    }
+    catch (const std::exception& e)
+    {
+        throw;
+    }
```

Comment:
Here, you're just catching any exception and are rethrowing it. That is the same as not catching it at all. What is the reason for this code?